pkg/chartutil/save.go

```diff
@@ -63,6 +63,12 @@ func SaveDir(c *chart.Chart, dest string) error {
 	// Save templates
 	for _, f := range c.Templates {
 		n := filepath.Join(outdir, f.Name)
+
+		d := filepath.Dir(n)
```

Comment:
why don't we just use `outdir` instead?

Reply:
I'm confused. Use `outdir` for what?

My understanding is that `outdir` is the base path of the chart, so it needs to be joined with the directory of each template name, before passing the joined path into `os.MkdirAll`.

This is basically doing the exact same thing that's done for `c.Files`, starting at line 72

Reply:
Oh I see, thank you. I forgot that `f.Name` can include directories in the name, hence the need to call `filepath.Dir` on that instead. My bad.